runtime/onert/core/src/util/EventRecorder.h

```diff
@@ -17,28 +17,75 @@
 #ifndef __ONERT_UTIL_EVENT_RECORDER_H__
 #define __ONERT_UTIL_EVENT_RECORDER_H__
 
+#include "util/TracingCtx.h"
+
 #include <map>
 #include <memory>
 #include <mutex>
 
 #include <vector>
 
+struct SubgDurationEvent;
+struct OpDurationEvent;
+
+class DurationEventVisitor
+{
+public:
+  virtual ~DurationEventVisitor() = default;
+
+  virtual std::string visit(const SubgDurationEvent &) const
+  {
+    throw std::runtime_error("Please implement");
+  }
+  virtual std::string visit(const OpDurationEvent &) const
+  {
+    throw std::runtime_error("Please implement");
+  }
+};
+
+// refer to https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit#
 struct Event
 {
-  std::string name;
```

Comment:
`name` and `tid` are different from type of writer(e.g., `ChromeTracingEventWriter`). So now deciding `name` and `tid` is specific writer's role. Thus, they are removed from here.